pkg/controller/postgressnapshot/postgressnapshot_controller.go

```diff
@@ -0,0 +1,226 @@
+func newReconciler(mgr manager.Manager) reconcile.Reconciler {
+	logger := logrus.WithFields(logrus.Fields{"controller": "controller_postgres_snapshot"})
+	return &ReconcilePostgresSnapshot{
+		client:            mgr.GetClient(),
+		scheme:            mgr.GetScheme(),
+		logger:            logger,
+		ConfigManager:     croAws.NewDefaultConfigMapConfigManager(mgr.GetClient()),
+		CredentialManager: croAws.NewCredentialMinterCredentialManager(mgr.GetClient()),
+	}
+}
+
+// add adds a new Controller to mgr with r as the reconcile.Reconciler
+func add(mgr manager.Manager, r reconcile.Reconciler) error {
+	// Create a new controller
+	c, err := controller.New("postgressnapshot-controller", mgr, controller.Options{Reconciler: r})
+	if err != nil {
+		return err
+	}
+
+	// Watch for changes to primary resource PostgresSnapshot
+	err = c.Watch(&source.Kind{Type: &integreatlyv1alpha1.PostgresSnapshot{}}, &handler.EnqueueRequestForObject{})
+	if err != nil {
+		return err
+	}
+
+	// Watch for changes to secondary resource Pods and requeue the owner PostgresSnapshot
+	err = c.Watch(&source.Kind{Type: &corev1.Pod{}}, &handler.EnqueueRequestForOwner{
+		IsController: true,
+		OwnerType:    &integreatlyv1alpha1.PostgresSnapshot{},
+	})
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// blank assignment to verify that ReconcilePostgresSnapshot implements reconcile.Reconciler
+var _ reconcile.Reconciler = &ReconcilePostgresSnapshot{}
+
+// ReconcilePostgresSnapshot reconciles a PostgresSnapshot object
+type ReconcilePostgresSnapshot struct {
+	// This client, initialized using mgr.Client() above, is a split client
+	// that reads objects from the cache and writes to the apiserver
+	client            client.Client
+	scheme            *runtime.Scheme
+	logger            *logrus.Entry
+	ConfigManager     croAws.ConfigManager
+	CredentialManager croAws.CredentialManager
+}
+
+// Reconcile reads that state of the cluster for a PostgresSnapshot object and makes changes based on the state read
+// and what is in the PostgresSnapshot.Spec
+func (r *ReconcilePostgresSnapshot) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	r.logger.Info("reconciling postgres snapshot")
+	ctx := context.TODO()
+
+	// Fetch the PostgresSnapshot instance
+	instance := &integreatlyv1alpha1.PostgresSnapshot{}
+	err := r.client.Get(context.TODO(), request.NamespacedName, instance)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
+			// Return and don't requeue
+			return reconcile.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		return reconcile.Result{}, err
+	}
+
+	// check status, if complete return
+	if instance.Status.Phase == croType.PhaseComplete {
+		r.logger.Infof("found existing rds snapshot for %s", instance.Name)
```

Comment:
I think this could cause some confusion when reading the logs. We haven't found an existing snapshot here, we're skipping creation of a snapshot as the phase is complete. I think we could change the wording here to something like that.

I think this check is fine to keep as we cannot create a snapshot from a previous time/date so if a user goes into AWS and removes the snapshot, we cannot really recreate it.

It would be nice if we could report on a snapshot for this time having been cleaned up and have a status to reflect that. But I don't think that was part of this ticket.